packages/core/contracts/test/ValidationsNatspec.sol

```diff
@@ -32,18 +32,18 @@ contract HasStateVariableAssignmentNatspec3 {
   uint y = 2;
 }
 
-/// @custom:oz-upgrades-unsafe-allow state-variable-immutable state-variable-assignment
+/// @custom:oz-upgrades-unsafe-allow state-variable-immutable
```

Comment:
shouldn't we remove the natspec completelly for immutable assignment.

`uint immutable x;` deserves an error, but `uint immutable x = 1;` should't require anything to be accepted

Reply:
It's a little more complicated than that IMO. Your example is good, but I don't think we should blindly accept the following:

```
    uint immutable x = block.number;
```

Although this makes me think the error in this case is indeed the assignment, because the user should be aware that it will happen in the implementation and not the proxy. Maybe we should rename `state-variable-assignment` to `implementation-assignment` or something like that.

Reply:
To be clear: I do want us to suport `uint immutable x = 1` but we'll need to make sure right hand side is a "pure" value.

Reply:
to me it is clear that `uint immutable x = block.number;` will take the block.number at construction of the implementation. I think this is the entire idea, and it should be authorized.

It's a matter of "who the users are" ... this is a more advanced usecase that require some expertise. You may argue that we need to warn novice user about this, but I think novice user won't go anywhere near immutable anyway.

Reply:
I don't agree, I've seen people struggle to understand the interaction between immutables and proxies, and I definitely expect novice users to try to use them.